src/main/flight/mixer.c

```diff
@@ -269,83 +273,123 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
-        const float stickDeflectionYawAbs = getRcDeflectionAbs(FD_YAW);
-
-        const float stickDeflectionPitchExpo = flipPowerFactor * stickDeflectionPitchAbs + power3(stickDeflectionPitchAbs) * (1 - flipPowerFactor);
-        const float stickDeflectionRollExpo = flipPowerFactor * stickDeflectionRollAbs + power3(stickDeflectionRollAbs) * (1 - flipPowerFactor);
-        const float stickDeflectionYawExpo = flipPowerFactor * stickDeflectionYawAbs + power3(stickDeflectionYawAbs) * (1 - flipPowerFactor);
-
-        float signPitch = getRcDeflection(FD_PITCH) < 0 ? 1 : -1;
-        float signRoll = getRcDeflection(FD_ROLL) < 0 ? 1 : -1;
-        float signYaw = (getRcDeflection(FD_YAW) < 0 ? 1 : -1) * (mixerConfig()->yaw_motors_reversed ? 1 : -1);
-
-        float stickDeflectionLength = sqrtf(sq(stickDeflectionPitchAbs) + sq(stickDeflectionRollAbs));
-        float stickDeflectionExpoLength = sqrtf(sq(stickDeflectionPitchExpo) + sq(stickDeflectionRollExpo));
-
-        if (stickDeflectionYawAbs > MAX(stickDeflectionPitchAbs, stickDeflectionRollAbs)) {
-            // If yaw is the dominant, disable pitch and roll
-            stickDeflectionLength = stickDeflectionYawAbs;
-            stickDeflectionExpoLength = stickDeflectionYawExpo;
-            signRoll = 0;
+#ifdef USE_ACC
+    static bool isFirstTiltAngleRead = true;
+    static float tiltAngleAtStart = 1.0f;
+#endif
+
+    if (!isCrashFlipModeActive()) {
+#ifdef USE_ACC
+        // trigger the capture of initial tilt angle on next activation of crashflip mode
+        isFirstTiltAngleRead = true;
+#endif
+        // signal that crashflip mode is off
+        return false;
+    }
+
+    const float stickDeflectionPitchAbs = getRcDeflectionAbs(FD_PITCH);
+    const float stickDeflectionRollAbs = getRcDeflectionAbs(FD_ROLL);
+    const float stickDeflectionYawAbs = getRcDeflectionAbs(FD_YAW);
+
+    float signPitch = getRcDeflection(FD_PITCH) < 0 ? 1 : -1;
+    float signRoll = getRcDeflection(FD_ROLL) < 0 ? 1 : -1;
+    float signYaw = (getRcDeflection(FD_YAW) < 0 ? 1 : -1) * (mixerConfig()->yaw_motors_reversed ? 1 : -1);
+
+    float stickDeflectionLength = sqrtf(sq(stickDeflectionPitchAbs) + sq(stickDeflectionRollAbs));
+
+    if (stickDeflectionYawAbs > MAX(stickDeflectionPitchAbs, stickDeflectionRollAbs)) {
+        // If yaw is the dominant, disable pitch and roll
+        stickDeflectionLength = stickDeflectionYawAbs;
+        signRoll = 0;
+        signPitch = 0;
+    } else {
+        // If pitch/roll dominant, disable yaw
+        signYaw = 0;
+    }
+
+    const float cosPhi = (stickDeflectionLength > 0) ? (stickDeflectionPitchAbs + stickDeflectionRollAbs) / (sqrtf(2.0f) * stickDeflectionLength) : 0;
+    const float cosThreshold = sqrtf(3.0f)/2.0f; // cos(PI/6.0f)
+
+    if (cosPhi < cosThreshold) {
+        // Enforce either roll or pitch exclusively, if not on diagonal
+        if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
```

Comment:
I thought it would be OK like that, and it seems OK... what was your concern?

Reply:
Hmm OK  I see. 
In practice, I think it's OK as is, though perhaps a more expert refactoring could be done later.
I say that because although some people may move the stick slowly, the vast majority just hammer the stick to the full position.
Minor variances in calculation of stick position are not that important afaik.
And if for instance the user shifts, say, from a deflection on roll, to one that is containing significant pitch deflection, then abruptly the behaviour changes to a single motor only.
There's no subtlety in this code ;-)
So I think it can be left as is - unless it could be simplified or made significantly more efficient.

Reply:
(these coding 'imperfections' that are not visible from user perspective are making refactoring and maintenance harder - it is harder to understand what is the code doing)

Reply:
I agree.  At this point, though, let's let sleeping dogs lie?